thentos-adhocracy/src/Thentos/Adhocracy3/Backend/Api/Simple.hs

```diff
@@ -403,17 +417,26 @@ login r = AC.logIfError'P $ do
 createUserInA3'P :: UserFormData -> AC.Action DB UserId
 createUserInA3'P user = do
     config <- AC.getConfig'P
-    let a3req = fromMaybe (error "createUserInA3'P: mkUserCreationRequestForA3 failed, check config!") $
+    let a3req = fromMaybe
+                (error "createUserInA3'P: mkUserCreationRequestForA3 failed, check config!") $
                 mkUserCreationRequestForA3 config user
     a3resp <- liftLIO . ioTCB . sendRequest $ a3req
     when (responseCode a3resp >= 400) $ do
         throwError . A3BackendErrorResponse (responseCode a3resp) $ Client.responseBody a3resp
     extractUserId a3resp
   where
-    sendRequest ::  Client.Request -> IO (Client.Response LBS)
-    sendRequest req = Client.withManager Client.defaultManagerSettings $ Client.httpLbs req
     responseCode = Status.statusCode . Client.responseStatus
 
+-- | Activate a user in A3 and return the response received from A3.
+activateUserInA3'P :: ActivationRequest -> AC.Action DB RequestResult
+activateUserInA3'P actReq = do
+    config <- AC.getConfig'P
+    let a3req = fromMaybe (error "activateUserInA3'P: mkRequestForA3 failed, check config!") $
```

Comment:
If I understand correctly, this error could happen either because (1) the config file is wrong, (2) A3 is down. I think the correct http status code in the latter case is 502, not 500.